exercises/circular-buffer/src/lib.rs

```diff
@@ -1 +1,44 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
 
+pub struct CircularBuffer<T: Debug> {
+    // This field is here to make the template compile and not to
+    // complain about unused type parameter 'T'. Once you start
+    // solving the exercise, delete this field and the 'std::marker::PhantomData'
+    // import.
+    field: PhantomData<T>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    EmptyBuffer,
+    FullBuffer,
+}
+
+impl<T: Debug> CircularBuffer<T> {
+    pub fn new(capacity: usize) -> Self {
+        unimplemented!(
+            "Construct a new CircularBuffer with the capacity to hold {}.",
+            match capacity {
+                1 => format!("1 element"),
+                _ => format!("{} elements", capacity),
+            }
+        );
+    }
+
+    pub fn write(&mut self, element: T) -> Result<(), Error> {
```

Comment:
I think `Result<(), Error>` makes sense and is the accepted way, as https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all does this as well. While it would be possible to return the index of the inserted value, I think the API presented in this exercise is not conducive to a client of this code being able to do anything with that information, so I wouldn't add it.

Reply:
Agree: `Result<(), Error>` is fine and idiomatic.